code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,261 @@
+		T.date = worlddate2text()
+		T.time = worldtime2text()
+		SSeconomy.add_transaction_log(customer_account,T)
+
+
+		print_receipt()
+		playsound(loc, 'sound/items/polaroid1.ogg', 50, 1)
+		to_chat(user, span("notice", "Transaction to [destinationact] account approved for a total of [sum] credits."))
+
+		sum = 0
+		receipt = ""
+
+// VUEUI Below <3
+
+
+/obj/item/device/nanoquikpay/attack_self(mob/user as mob)
+	var/datum/vueui/ui = SSvueui.get_open_ui(usr, src)
+	if (!ui)
+		ui = new(usr, src, "quikpay-main", 400, 400, "NT Quik-Pay")
+	ui.open()
+
+/obj/item/device/nanoquikpay/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	if(!data)
+		. = data = list()
+
+	VUEUI_SET_CHECK_IFNOTSET(data["items"], items, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["price"], items, ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["tmp_name"], "", ., data)
+	VUEUI_SET_CHECK_IFNOTSET(data["tmp_price"], 0, ., data)
+	VUEUI_SET_CHECK(data["tmp_price"], max(0, data["tmp_price"]), ., data)
+	if(data["tmp_price"] < 0)
+		data["tmp_price"] = 0
+		. = data
+	VUEUI_SET_CHECK_IFNOTSET(data["selection"], list("_" = 0), ., data)
+	VUEUI_SET_CHECK(data["editmode"], editmode, ., data)
+	VUEUI_SET_CHECK(data["destinationact"], destinationact, ., data)
+
+/obj/item/device/nanoquikpay/Topic(href, href_list)
+	var/datum/vueui/ui = href_list["vueui"]
+	if(!istype(ui))
+		return
+	if(href_list["add"])
+
+		if(editmode == 0)
+			to_chat(src, span("notice", "You don't have access to use this option."))
+			return 0
+
+		items[href_list["add"]["name"]] = href_list["add"]["price"]
+		ui.data["items"][href_list["add"]["name"]] = href_list["add"]["price"]
+		. = TRUE
+	if(href_list["remove"])
+
+		if(editmode == 0)
+			to_chat(src, span("notice", "You don't have access to use this option."))
+			return 0
+		items -= href_list["remove"]
+		ui.data["items"] -= href_list["remove"]
+		. = TRUE
+	if(href_list["confirm"])
+		var/selection = ui.data["selection"]
+		var/items = ui.data["items"]
+		for(var/name in selection)
+			if(items[name] && selection[name])
+				sum += items[name] * selection[name]
+				receipt += "<b>[name]</b> : [items[name]]x[selection[name]]:  [items[name] * selection[name]]<br>"
+		ui.activeui = "quikpay-confirmation"	
+		. = TRUE
+	if(href_list["return"])
+		sum = 0
+		receipt = ""
+		ui.activeui = "quikpay-main"	
+		. = TRUE
+
+
+	if(href_list["locking"])
+		if(editmode == 1)
+			editmode = 0
+			to_chat(src, span("notice", "Device Locked."))
+			SSvueui.check_uis_for_change(src)
+			return 0
+		if(editmode == 0)
+			var/attempt_code = input("Enter the edit code", "Confirm edit access code") as num //Copied the eftpos method, dont judge
+			if(attempt_code == access_code)
+				editmode = 1
+				to_chat(src, span("notice", "Device Unlocked."))
+				SSvueui.check_uis_for_change(src)
+		. = TRUE
+		
+
+	if(href_list["accountselect"])
+
+		if(editmode == 0)
+			to_chat(src, span("notice", "You don't have access to use this option."))
```

Comment:
Silent failure should be OK. As usual operation would never result this line to be called, UI should hide buttons that lead to this topic calling. This is here to check for hackers that have modified UI on there side and tries to call topics that shouldn't be available.